src/java/components/hu.bme.mit.inf.modes3.components.dashboard/src/main/webapp/public_html/js/class/locomotive.js

```diff
@@ -1,70 +1,160 @@
+
+	// add svg representation as well
+	this.svgElemGroup = $('<g />').attr('id', this.config.svgGroup);
+
+	// adding position circle
+	this.svgElemPosition = $('<circle />').attr({
+		id : 'position',
+		style : window.settings.locomotiveCircleStyle,
+		cx : 0,
+		cy : 0,
+		r : 100.91698
+	});
+	this.svgTitle = $('<text />').attr({
+		id : "title",
+		style : window.settings.locomotiveTextStyle,
+		x : 120,
+		y : 0
+	}).append($('<tspan />').text(this.config.name));
+	this.svgSpeedText = $('<text />').attr({
+		id : "title",
+		style : window.settings.locomotiveSpeedTextStyle,
+		x : 120,
+		y : 120
+	}).append($('<tspan />').text('0.00 m/s'));
+	//	
+	// adding background to all of these
+	var background = $('<rect />').attr({
+		id : "bg-" + this.config.svgGroup,
+		x : 115,
+		y : -125,
+		width : 0,
+		height : 0,
+		style : "fill:#FFF;fill-opacity:0.6;"
+	});
+
+	this.svgElemGroup.attr('transform', 'translate(100,100)');
+	this.svgElemGroup.append(background);
+	this.svgElemGroup.append(this.svgElemPosition);
+	this.svgElemGroup.append(this.svgTitle);
+	this.svgElemGroup.append(this.svgSpeedText);
+
+	// adding svgElement to it parent
+	$('#layer4').append(this.svgElemGroup);
+};
+
+LocomotiveController.prototype.positionInformationReceived = function(info) {
+	this.position = info.realposition;
+
+	// transform coordinates to the dimensions of the viewbox
+	// also, positions are received in cm, but the SVG is designed with mm range
+	// thus it need's to be multipled by 10
+	this.position.x *= 10 * 8858.2677 / 2500.0;
+	this.position.y *= 10 * 4960.63 / 1400.0;
+
+	var pos = 'translate(' + this.position.x + ',' + this.position.y + ')';
+	this.svgElemGroup.attr('transform', pos);
+
+	// CV is sending position information with a fix rate even if objects are
+	// not tracked at the moment. If that occurs, we show on the UI with
+	// different style
+	if (info.tracked[0]) {
+		this.svgElemPosition.attr('style', window.settings.locomotiveCircleStyle);
+	} else {
+		this.svgElemPosition.attr('style', window.settings.locomotiveNotTrackedCircleStyle);
+	}
+};
+
+LocomotiveController.prototype.pushSpeed = function(speed) {
 	if (speed < 0) {
-        log("Pushing train speed: "+speed);
-        this.controller.pushTrainSpeed(this.config.address, Math.abs(speed), 1);
+		log("Pushing train speed: " + speed);
+		this.controller.pushTrainSpeed(this.config.address, Math.abs(speed), 1);
 
-    } else {
-    	log("Pushing train speed: "+speed);
-    	this.controller.pushTrainSpeed(this.config.address, speed, 0);
-    }
+	} else {
+		log("Pushing train speed: " + speed);
+		this.controller.pushTrainSpeed(this.config.address, speed, 0);
+	}
 };
 
-LocomotiveController.prototype.setSpeed = function (speed) {
-    this.speed = speed;
+LocomotiveController.prototype.setSpeed = function(speed) {
+	this.speed = speed;
 	this.rangeInput.val(speed);
-	this.indicatorElem.text(speed);
+	
+	if( speed < 0 ) {
+		this.indicatorElem.text((speed/48*-100)+"% (R)");
```

Comment:
What does this computation do? Only thing I can see is magic constants :D

Reply:
If speed is a negative number (in case of reverse movement), show positive number and an R letter instead of a negative number. Also, maximum speed of the trains are defined to be 48 (previously it was 50), and makes a percentage value based on the full speed.

Maximum speed constant should be equivalent with the values defined in touchboard project though.